core/lib/include/debug.h

```diff
@@ -44,16 +44,19 @@ extern "C" {
  */
 #ifdef DEVELHELP
 #include "cpu_conf.h"
-#define DEBUG_PRINT(...) \
+#define DEBUG_PRINT(X, ...) \
```

Comment:
That's not standard compliant, as the C standard requires at least on argument to be present for the `...`.

Reply:
https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3096.pdf p175:

If there is a ... in the identifier-list in the macro definition, then the trailing arguments **(if any)**,
including any separating comma preprocessing tokens, are merged to form a single item: the variable
arguments.

Reply:
```
<source>: .. warning: no previous prototype for function 'foo' [-Wmissing-prototypes] 
```
are you sure that is a applicable warning level ?

Reply:
```
#define DEBUG(X, ...) \
    do { \
        if (sizeof((const char[]){"empty"#__VA_ARGS__}) == sizeof((const char[]){"empty"}))\
            printf(X __VA_OPT__(,) __VA_ARGS__);\
        else {\
            if ((thread_get_active() == NULL) || \
                (thread_get_active()->stack_size >= \
                THREAD_EXTRA_STACKSIZE_PRINTF)) { \
                    printf(X __VA_OPT__(,) __VA_ARGS__); \
                } \
            else { \
                puts("Cannot debug, stack too small. Consider using DEBUG_PUTS()."); \
            } }\
    } while (0)
 ```
 
 would be the solution following the c2x draft but clang doesn't like that  either :
 
 ```
 -Os -std=c2x -Weverything
 
     Output of x86-64 clang 17.0.1 (Compiler #1)

<source>:32:20: warning: must specify at least one argument for '...' parameter of variadic macro [-Wgnu-zero-variadic-macro-arguments]
   32 |     DEBUG("Hallo\n");
 ```
 seems like clangs warnings are not up to date